File: 2022/py/day1_p2.py
with open('../day1.txt') as f:
    # get every line of the file in a list
    content = f.readlines()

    # start at beginning of list and add nums to var until '\n'
    nums = 0
    groups = []
    for num in content:
        if num != '\n':
            nums += int(num)
        else:
            # reset the group count
            groups.append(nums)
            nums = 0

groups.sort(reverse=True)
print(groups[0] + groups[1] + groups[2])

File: 2022/js/day1_p2.js
// Read the file and convert to an array
const fs = require('fs');
const content = fs.readFileSync('../day1.txt', 'utf8').split('\n');

let nums = 0;
let groups = [];

for (let i = 0; i < content.length; ++i) {
    if (content[i] !== '') {
        nums += Number(content[i]);
    } else {
        groups.push(nums);
        nums = 0;
    }
}
// sort the array in reverse order
groups.sort();
groups.reverse();
console.log(groups[0] + groups[1] + groups[2]);
